app/javascript/packages/document-capture/hooks/use-steps.ts

```diff
@@ -0,0 +1,94 @@
+import { useContext } from 'react';
+import type { FormStep } from '@18f/identity-form-steps';
+import { useI18n } from '@18f/identity-react-i18n';
+import { UploadFormEntriesError } from '../services/upload';
+import DocumentsStep from '../components/documents-step';
+import InPersonPrepareStep from '../components/in-person-prepare-step';
+import InPersonLocationPostOfficeSearchStep from '../components/in-person-location-post-office-search-step';
+import InPersonLocationFullAddressEntryPostOfficeSearchStep from '../components/in-person-location-full-address-entry-post-office-search-step';
+import InPersonSwitchBackStep from '../components/in-person-switch-back-step';
+import ReviewIssuesStep from '../components/review-issues-step';
+import withProps from '../higher-order/with-props';
+import { InPersonContext } from '../context';
+import UploadContext from '../context/upload';
+
+const useReviewStep = (submissionError: Error | undefined) => {
+  const { t } = useI18n();
+  return {
+    name: 'review',
+    form:
+      submissionError instanceof UploadFormEntriesError
+        ? withProps({
+            remainingAttempts: submissionError.remainingAttempts,
+            isFailedResult: submissionError.isFailedResult,
+            isFailedDocType: submissionError.isFailedDocType,
+            captureHints: submissionError.hints,
+            pii: submissionError.pii,
+            failedImageFingerprints: submissionError.failed_image_fingerprints,
+          })(ReviewIssuesStep)
+        : ReviewIssuesStep,
+    title: t('errors.doc_auth.rate_limited_heading'),
+  };
+};
+
+export const useSteps = (submissionError: Error | undefined) => {
```

Comment:
Can we add a return type here? My instinct from the previous code is that this should be returning `FormStep[]`, but looking at the logic of this function I think we may also be returning `undefined` in some cases. Is that expected?

```suggestion
export const useSteps = (submissionError: Error | undefined): FormStep[] => {
```